clang/test/CIR/Transforms/mem2reg.cir

```diff
@@ -0,0 +1,206 @@
+// RUN: cir-opt %s -cir-flatten-cfg -mem2reg -o - | FileCheck %s
+
```

Comment:
Nice initial set of testcases. I suggest something slightly different:
- Make this test only cover the CIR version of `return_42` here, remove the rest.
- Use all the C/C++ code in this example in an actual source file, and invoke clang to test them. Include source version of `return_42` as well. Check both CIR outputs using pass-before and pass-after (something similar to `clang/test/CIR/CodeGen/array-init-destroy.cpp`). This way you also test the driver flag you added.

Reply:
done!